webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -46,24 +82,39 @@ export default {
           variables,
         })
         if (!(markAsRead && markAsRead.read === true)) return
-        this.notifications = this.notifications.map(n => {
-          return n.from.id === markAsRead.from.id ? markAsRead : n
+        this.displayedNotifications = this.displayedNotifications.map(n => {
+          return this.equalNotification(n, markAsRead) ? markAsRead : n
         })
       } catch (err) {
         throw new Error(err)
       }
     },
+    equalNotification(a, b) {
```

Comment:
~I would not register this on the component if it's not being used in the template or elsewhere. You can assign the function in another function, see above :point_up:~

What I don't understand: Why is the check for `id` not enough?

Reply:
Somebody can mention me in the same post again by editing. Then a new notification with the same `from.id` is created. But the first one can be still in the list and be read already.
The second unread one will not appear if you check only for `from.id `, even though it is in the database as separate notification with an other `read` value and a different `createdAt`.